--- /users/students/r0678912/Downloads/zigzag/hls_impl/PE/Catapult/SIF/power_top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_pa000000.vhdl	2021-07-13 11:47:37.149352146 +0200
+++ /users/students/r0678912/Downloads/zigzag/hls_impl/PE/Catapult/SIF/calypto_opt_vhdl/iter_write_rtl/calypto_insert_obs/2/power_top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_pa000000.vhdl	2021-07-13 11:48:16.540930338 +0200
@@ -1721,6 +1721,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000031
     IS
   PORT(
@@ -1735,6 +1737,8 @@
   );
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000031;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000031
     IS
   -- Default Constants
@@ -1744,6 +1748,12 @@
   SIGNAL W_wr_data_rsci_ivld_bfwt : STD_LOGIC;
   SIGNAL W_wr_data_rsci_idat_gtd : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL W_wr_data_rsci_idat_bfwt : STD_LOGIC_VECTOR (15 DOWNTO 0);
+  --PowerPro-CG
+  signal rst_1 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal W_wr_data_rsci_ivld_1 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal cg_o_W_wr_data_rsci_idat_bfwt_en : ieee.std_logic_1164.std_logic_vector(0 downto 0);
 
   FUNCTION MUX_s_1_2_2(input_0 : STD_LOGIC;
   input_1 : STD_LOGIC;
@@ -1784,6 +1794,12 @@
       W_wr_data_rsci_idat, W_wr_data_rsci_ivld);
   W_wr_data_rsci_idat_mxwt <= MUX_v_16_2_2(W_wr_data_rsci_idat_gtd, W_wr_data_rsci_idat_bfwt,
       W_wr_data_rsci_bcwt);
+  --PowerPro-CG
+  inst_cg_obs_topless_5comma_32comma_1comm : cg_obs_topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000031 port map (
+              rst => rst_1,
+              W_wr_data_rsci_ivld => W_wr_data_rsci_ivld_1,
+              W_wr_data_rsci_idat_bfwt_en => cg_o_W_wr_data_rsci_idat_bfwt_en
+           );
   PROCESS (clk)
   BEGIN
     IF clk'EVENT AND ( clk = '1' ) THEN
@@ -1795,15 +1811,23 @@
       END IF;
     END IF;
   END PROCESS;
+  --PowerPro-CG
+  rst_1 <= (rst);
+  --PowerPro-CG
+  W_wr_data_rsci_ivld_1 <= (W_wr_data_rsci_ivld);
   PROCESS (clk)
   BEGIN
     IF clk'EVENT AND ( clk = '1' ) THEN
       IF (rst = '1') THEN
         W_wr_data_rsci_ivld_bfwt <= '0';
+       if ( calypto_lib.powerpro_cg_package.is_enabled(cg_o_W_wr_data_rsci_idat_bfwt_en) ) then --PowerPro-CG
         W_wr_data_rsci_idat_bfwt <= STD_LOGIC_VECTOR'( "0000000000000000");
+       end if;
       ELSIF ( W_wr_data_rsci_biwt = '1' ) THEN
         W_wr_data_rsci_ivld_bfwt <= W_wr_data_rsci_ivld;
+         if ( calypto_lib.powerpro_cg_package.is_enabled(cg_o_W_wr_data_rsci_idat_bfwt_en) ) then --PowerPro-CG
         W_wr_data_rsci_idat_bfwt <= W_wr_data_rsci_idat_gtd;
+       end if;
       END IF;
     END IF;
   END PROCESS;
@@ -3733,6 +3757,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000008
     IS
   PORT(
@@ -3749,6 +3775,8 @@
   );
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000008;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000008
     IS
   -- Default Constants
@@ -3771,7 +3799,7 @@
       W_wr_data_rsci_bdwt : OUT STD_LOGIC
     );
   END COMPONENT;
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000031
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000031
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -3783,6 +3811,20 @@
       W_wr_data_rsci_idat : IN STD_LOGIC_VECTOR (15 DOWNTO 0)
     );
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000031
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      W_wr_data_rsci_ivld_mxwt : OUT STD_LOGIC;
+--      W_wr_data_rsci_idat_mxwt : OUT STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_wr_data_rsci_biwt : IN STD_LOGIC;
+--      W_wr_data_rsci_bdwt : IN STD_LOGIC;
+--      W_wr_data_rsci_ivld : IN STD_LOGIC;
+--      W_wr_data_rsci_idat : IN STD_LOGIC_VECTOR (15 DOWNTO 0)
+--    );
+--  END COMPONENT;
+                
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_W_wr_data_rsci_W_wr_data_wait_dp_inst_W_wr_data_rsci_idat_mxwt
       : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_W_wr_data_rsci_W_wr_data_wait_dp_inst_W_wr_data_rsci_idat
@@ -5041,6 +5083,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
     IS
   PORT(
@@ -5073,6 +5117,8 @@
   );
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
     IS
   -- Default Constants
@@ -6272,7 +6318,7 @@
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_I_wr_data_rsci_inst_I_wr_data_rsci_idat_mxwt
       : STD_LOGIC_VECTOR (15 DOWNTO 0);
 
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000008
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000008
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -6286,6 +6332,22 @@
       W_wr_data_rsci_idat_mxwt : OUT STD_LOGIC_VECTOR (15 DOWNTO 0)
     );
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000008
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      W_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_wr_data_rsc_vld : IN STD_LOGIC;
+--      W_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      run_wen : IN STD_LOGIC;
+--      run_wten : IN STD_LOGIC;
+--      W_wr_data_rsci_oswt : IN STD_LOGIC;
+--      W_wr_data_rsci_ivld_mxwt : OUT STD_LOGIC;
+--      W_wr_data_rsci_idat_mxwt : OUT STD_LOGIC_VECTOR (15 DOWNTO 0)
+--    );
+--  END COMPONENT;
+                
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_W_wr_data_rsci_inst_W_wr_data_rsc_dat
       : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_W_wr_data_rsci_inst_W_wr_data_rsci_idat_mxwt
@@ -6380,6 +6442,14 @@
   END COMPONENT;
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_run_fsm_inst_fsm_output
       : STD_LOGIC_VECTOR (1 DOWNTO 0);
+  --PowerPro-CG
+  signal rst_1 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_run_fsm_inst_fsm_output_1 : ieee.std_logic_1164.std_logic_vector(1 downto 0);
+  --PowerPro-CG
+  signal land_9_lpi_1_dfm_1_2 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal cg_o_O_write_data_data_sva_en : ieee.std_logic_1164.std_logic_vector(0 downto 0);
 
   FUNCTION CONV_SL_1_1(input_val:BOOLEAN)
   RETURN STD_LOGIC IS
@@ -7693,6 +7763,13 @@
   mux_662_nl <= MUX_s_1_2_2((fsm_output(1)), reg_W_instr_in_rsci_oswt_cse_1, and_526_cse);
   mux_661_nl <= MUX_s_1_2_2((fsm_output(1)), nor_tmp_217, and_526_cse);
   mux_663_itm <= MUX_s_1_2_2(mux_662_nl, mux_661_nl, or_756_cse);
+  --PowerPro-CG
+  inst_cg_obs_topless_5comma_32comma_1comm : cg_obs_topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003 port map (
+              rst => rst_1,
+              fsm_output => rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_run_fsm_inst_fsm_output_1,
+              O_write_data_data_sva_en => cg_o_O_write_data_data_sva_en,
+              land_9_lpi_1_dfm_1 => land_9_lpi_1_dfm_1_2
+           );
   PROCESS (clk)
   BEGIN
     IF clk'EVENT AND ( clk = '1' ) THEN
@@ -8864,9 +8941,13 @@
   BEGIN
     IF clk'EVENT AND ( clk = '1' ) THEN
       IF (rst = '1') THEN
+       if ( calypto_lib.powerpro_cg_package.is_enabled(cg_o_O_write_data_data_sva_en) ) then --PowerPro-CG
         O_write_data_data_sva <= STD_LOGIC_VECTOR'( "0000000000000000");
+       end if;
       ELSIF ( (land_9_lpi_1_dfm_1_1 AND run_wen) = '1' ) THEN
+         if ( calypto_lib.powerpro_cg_package.is_enabled(cg_o_O_write_data_data_sva_en) ) then --PowerPro-CG
         O_write_data_data_sva <= O_write_data_data_sva_mx0;
+       end if;
       END IF;
     END IF;
   END PROCESS;
@@ -9437,6 +9518,12 @@
   and_549_nl <= if_5_and_svs_1 AND (fsm_output(1));
   skid_buf_top_pop_1_mux_1_nl <= MUX_v_2_2_2(skid_buf_top_cnt_sva, skid_buf_top_cnt_sva_1_1,
       and_549_nl);
+  --PowerPro-CG
+  rst_1 <= (rst);
+  --PowerPro-CG
+  rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_run_fsm_inst_fsm_output_1 <= (rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_run_fsm_inst_fsm_output);
+  --PowerPro-CG
+  land_9_lpi_1_dfm_1_2 <= (land_9_lpi_1_dfm_1);
   z_out_3 <= STD_LOGIC_VECTOR(CONV_UNSIGNED(UNSIGNED(skid_buf_top_pop_1_mux_1_nl)
       + UNSIGNED'( "11"), 2));
 END v2;
@@ -10125,6 +10212,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
     IS
   PORT(
@@ -10157,11 +10246,13 @@
   );
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
     IS
   -- Default Constants
 
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -10191,6 +10282,38 @@
       W_instr_in_rsc_rdy : OUT STD_LOGIC
     );
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      O_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_wr_data_rsc_vld : IN STD_LOGIC;
+--      O_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      O_rd_data_rsc_dat : OUT STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_rd_data_rsc_vld : OUT STD_LOGIC;
+--      O_rd_data_rsc_rdy : IN STD_LOGIC;
+--      I_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      I_wr_data_rsc_vld : IN STD_LOGIC;
+--      I_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      W_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_wr_data_rsc_vld : IN STD_LOGIC;
+--      W_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      wr_data_zero_guard_rsc_dat : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_vld : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_rdy : OUT STD_LOGIC;
+--      O_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      O_instr_in_rsc_vld : IN STD_LOGIC;
+--      O_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      I_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      I_instr_in_rsc_vld : IN STD_LOGIC;
+--      I_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      W_instr_in_rsc_vld : IN STD_LOGIC;
+--      W_instr_in_rsc_rdy : OUT STD_LOGIC
+--    );
+--  END COMPONENT;
+                
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_inst_O_wr_data_rsc_dat
       : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_inst_O_rd_data_rsc_dat
@@ -10265,6 +10388,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
     IS
   PORT(
@@ -10308,6 +10433,8 @@
   );
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
     IS
   -- Default Constants
@@ -10390,7 +10517,7 @@
   SIGNAL CCInst_W_instr_L2_out_rsc_dat : STD_LOGIC_VECTOR (109 DOWNTO 0);
   SIGNAL CCInst_W_instr_L3_out_rsc_dat : STD_LOGIC_VECTOR (159 DOWNTO 0);
 
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -10420,6 +10547,38 @@
       W_instr_in_rsc_rdy : OUT STD_LOGIC
     );
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      O_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_wr_data_rsc_vld : IN STD_LOGIC;
+--      O_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      O_rd_data_rsc_dat : OUT STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_rd_data_rsc_vld : OUT STD_LOGIC;
+--      O_rd_data_rsc_rdy : IN STD_LOGIC;
+--      I_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      I_wr_data_rsc_vld : IN STD_LOGIC;
+--      I_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      W_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_wr_data_rsc_vld : IN STD_LOGIC;
+--      W_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      wr_data_zero_guard_rsc_dat : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_vld : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_rdy : OUT STD_LOGIC;
+--      O_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      O_instr_in_rsc_vld : IN STD_LOGIC;
+--      O_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      I_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      I_instr_in_rsc_vld : IN STD_LOGIC;
+--      I_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      W_instr_in_rsc_vld : IN STD_LOGIC;
+--      W_instr_in_rsc_rdy : OUT STD_LOGIC
+--    );
+--  END COMPONENT;
+                
   SIGNAL PE_O_wr_data_rsc_dat : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL PE_O_rd_data_rsc_dat : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL PE_I_wr_data_rsc_dat : STD_LOGIC_VECTOR (15 DOWNTO 0);
@@ -10568,6 +10727,8 @@
   );
 END top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000
     IS
   -- Default Constants
@@ -10576,7 +10737,7 @@
   -- Interconnect Declarations
   SIGNAL O_data_out_rsc_dat_data : STD_LOGIC_VECTOR (15 DOWNTO 0);
 
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -10617,6 +10778,49 @@
       layer_instruction_in_rsc_rdy : OUT STD_LOGIC
     );
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      O_data_in_rsc_dat_data : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_data_in_rsc_vld : IN STD_LOGIC;
+--      O_data_in_rsc_rdy : OUT STD_LOGIC;
+--      O_data_out_rsc_dat_data : OUT STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_data_out_rsc_vld : OUT STD_LOGIC;
+--      O_data_out_rsc_rdy : IN STD_LOGIC;
+--      I_data_in_rsc_dat_data : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      I_data_in_rsc_vld : IN STD_LOGIC;
+--      I_data_in_rsc_rdy : OUT STD_LOGIC;
+--      W_data_in_rsc_dat_data : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_data_in_rsc_vld : IN STD_LOGIC;
+--      W_data_in_rsc_rdy : OUT STD_LOGIC;
+--      zero_guard_in_rsc_dat : IN STD_LOGIC;
+--      zero_guard_in_rsc_vld : IN STD_LOGIC;
+--      zero_guard_in_rsc_rdy : OUT STD_LOGIC;
+--      layer_instruction_in_rsc_dat_relevancy : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound : IN STD_LOGIC_VECTOR (24 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_1 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_1 : IN STD_LOGIC_VECTOR (54 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_2 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_2 : IN STD_LOGIC_VECTOR (79 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_3 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_3 : IN STD_LOGIC_VECTOR (24 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_4 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_4 : IN STD_LOGIC_VECTOR (44 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_5 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_5 : IN STD_LOGIC_VECTOR (69 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_6 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_6 : IN STD_LOGIC_VECTOR (24 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_7 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_7 : IN STD_LOGIC_VECTOR (44 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_8 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_8 : IN STD_LOGIC_VECTOR (69 DOWNTO 0);
+--      layer_instruction_in_rsc_vld : IN STD_LOGIC;
+--      layer_instruction_in_rsc_rdy : OUT STD_LOGIC
+--    );
+--  END COMPONENT;
+                
   SIGNAL top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000002
       : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000003
