--- /users/students/r0678912/Downloads/zigzag/hls_impl/PE/Catapult/SIF/power_top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_pa000000.vhdl	2021-07-13 11:47:37.149352146 +0200
+++ /users/students/r0678912/Downloads/zigzag/hls_impl/PE/Catapult/SIF/calypto_opt_vhdl/iter_write_rtl/calypto_insert_stb_s/0/power_top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_pa000000.vhdl	2021-07-13 11:49:04.744851751 +0200
@@ -5041,6 +5041,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
     IS
   PORT(
@@ -5069,10 +5071,14 @@
     I_instr_in_rsc_rdy : OUT STD_LOGIC;
     W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
     W_instr_in_rsc_vld : IN STD_LOGIC;
-    W_instr_in_rsc_rdy : OUT STD_LOGIC
-  );
+    W_instr_in_rsc_rdy : OUT STD_LOGIC;
+    --PowerPro-CG
+    ppro_reset_CLOCKclk_out_port_15 : in ieee.std_logic_1164.std_logic_vector(0 downto 0) );
+    
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
     IS
   -- Default Constants
@@ -6173,6 +6179,52 @@
   SIGNAL if_9_mux_127_nl : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL skid_buf_top_pop_1_mux_1_nl : STD_LOGIC_VECTOR (1 DOWNTO 0);
   SIGNAL and_549_nl : STD_LOGIC;
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000153_PP_0
+    PORT (
+      loop_bound : IN STD_LOGIC_VECTOR(24 DOWNTO 0);
+      tile_sizes : IN STD_LOGIC_VECTOR(24 DOWNTO 0);
+      pntr_in : IN STD_LOGIC_VECTOR(4 DOWNTO 0);
+      pntr_out : OUT STD_LOGIC_VECTOR(4 DOWNTO 0);
+      irrel_at_max_out : OUT STD_LOGIC;
+      irrel_at_zero_out : OUT STD_LOGIC;
+      all_at_max_1_out : OUT STD_LOGIC;
+      counter_in : IN STD_LOGIC_VECTOR(24 DOWNTO 0);
+      counter_out : OUT STD_LOGIC_VECTOR(24 DOWNTO 0);
+      tile_bound_in : IN STD_LOGIC_VECTOR(24 DOWNTO 0);
+      tile_bound_out : OUT STD_LOGIC_VECTOR(24 DOWNTO 0);
+    --PowerPro-CG
+    counter_nand_2_nl_5 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    counter_or_nl_3 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    mux_14_cse_21 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    for_1_for_1_for_1_for_1_nor_tmp_22 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    and_dcpl_3_23 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    mux_tmp_25_24 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    and_dcpl_25 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    for_1_unequal_3_tmp_27 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    and_dcpl_64_28 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    counter_nand_1_nl_6 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    counter_nand_3_nl_7 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    counter_nand_nl_4 : out ieee.std_logic_1164.std_logic_vector(0 downto 0);
+    --PowerPro-CG
+    operator_5_false_1_acc_4_nl_5_5 : out ieee.std_logic_1164.std_logic_vector(5 downto 5);
+    --PowerPro-CG
+    operator_5_false_1_acc_1_nl_5_5 : out ieee.std_logic_1164.std_logic_vector(5 downto 5);
+    --PowerPro-CG
+    operator_5_false_1_acc_nl_5_5 : out ieee.std_logic_1164.std_logic_vector(5 downto 5) );
+      
+  END COMPONENT;
+                
   COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000153
     PORT (
       loop_bound : IN STD_LOGIC_VECTOR(24 DOWNTO 0);
@@ -6380,6 +6432,44 @@
   END COMPONENT;
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_run_fsm_inst_fsm_output
       : STD_LOGIC_VECTOR (1 DOWNTO 0);
+  --PowerPro-CG
+  signal counter_nand_2_nl : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal counter_or_nl : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal mux_14_cse : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal for_1_for_1_for_1_for_1_nor_tmp : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal and_dcpl_3 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal mux_tmp_25 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal and_dcpl_26 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal for_1_unequal_3_tmp : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal and_dcpl_64 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal counter_nand_1_nl : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal counter_nand_3_nl : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal counter_nand_nl : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal W_mac_pntr_and_cse_1 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal clk_1 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal rst_1 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal operator_5_false_1_acc_4_nl_5_5 : ieee.std_logic_1164.std_logic_vector(5 downto 5);
+  --PowerPro-CG
+  signal operator_5_false_1_acc_1_nl_5_5 : ieee.std_logic_1164.std_logic_vector(5 downto 5);
+  --PowerPro-CG
+  signal operator_5_false_1_acc_nl_5_5 : ieee.std_logic_1164.std_logic_vector(5 downto 5);
+  --PowerPro-CG
+  signal cg_s_O_mac_counter_4_sva_en : ieee.std_logic_1164.std_logic_vector(0 downto 0);
 
   FUNCTION CONV_SL_1_1(input_val:BOOLEAN)
   RETURN STD_LOGIC IS
@@ -6637,9 +6727,11 @@
   END;
 
 BEGIN
-  O_mac_pntr_cntInst_run_rg : topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000153
+  --PowerPro-CG
+  O_mac_pntr_cntInst_run_rg : topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000153_PP_0
     PORT MAP(
-      loop_bound => O_mac_pntr_cntInst_run_rg_loop_bound,
+      loop_bound => 
+                    O_mac_pntr_cntInst_run_rg_loop_bound,
       tile_sizes => O_mac_pntr_cntInst_run_rg_tile_sizes,
       pntr_in => O_mac_pntr_cntInst_run_rg_pntr_in,
       pntr_out => O_mac_pntr_cntInst_run_rg_pntr_out,
@@ -6649,8 +6741,38 @@
       counter_in => O_mac_pntr_cntInst_run_rg_counter_in,
       counter_out => O_mac_pntr_cntInst_run_rg_counter_out,
       tile_bound_in => O_mac_pntr_cntInst_run_rg_tile_bound_in,
-      tile_bound_out => O_mac_pntr_cntInst_run_rg_tile_bound_out
-    );
+      tile_bound_out =>                         O_mac_pntr_cntInst_run_rg_tile_bound_out,
+      --PowerPro-CG
+      counter_nand_2_nl_5 => counter_nand_2_nl,
+      --PowerPro-CG
+      counter_or_nl_3 => counter_or_nl,
+      --PowerPro-CG
+      mux_14_cse_21 => mux_14_cse,
+      --PowerPro-CG
+      for_1_for_1_for_1_for_1_nor_tmp_22 => for_1_for_1_for_1_for_1_nor_tmp,
+      --PowerPro-CG
+      and_dcpl_3_23 => and_dcpl_3,
+      --PowerPro-CG
+      mux_tmp_25_24 => mux_tmp_25,
+      --PowerPro-CG
+      and_dcpl_25 => and_dcpl_26,
+      --PowerPro-CG
+      for_1_unequal_3_tmp_27 => for_1_unequal_3_tmp,
+      --PowerPro-CG
+      and_dcpl_64_28 => and_dcpl_64,
+      --PowerPro-CG
+      counter_nand_1_nl_6 => counter_nand_1_nl,
+      --PowerPro-CG
+      counter_nand_3_nl_7 => counter_nand_3_nl,
+      --PowerPro-CG
+      counter_nand_nl_4 => counter_nand_nl,
+      --PowerPro-CG
+      operator_5_false_1_acc_4_nl_5_5 => operator_5_false_1_acc_4_nl_5_5,
+      --PowerPro-CG
+      operator_5_false_1_acc_1_nl_5_5 => operator_5_false_1_acc_1_nl_5_5,
+      --PowerPro-CG
+      operator_5_false_1_acc_nl_5_5 => operator_5_false_1_acc_nl_5_5 );
+      
   O_mac_pntr_cntInst_run_rg_loop_bound <= O_instr_in_crt_lpi_1_dfm_mx0(24 DOWNTO
       0);
   O_mac_pntr_cntInst_run_rg_tile_sizes <= O_instr_in_crt_lpi_1_dfm_mx0(49 DOWNTO
@@ -7693,6 +7815,29 @@
   mux_662_nl <= MUX_s_1_2_2((fsm_output(1)), reg_W_instr_in_rsci_oswt_cse_1, and_526_cse);
   mux_661_nl <= MUX_s_1_2_2((fsm_output(1)), nor_tmp_217, and_526_cse);
   mux_663_itm <= MUX_s_1_2_2(mux_662_nl, mux_661_nl, or_756_cse);
+  --PowerPro-CG
+  inst_cg_sym_stb_topless_5comma_32comma_1 : cg_sym_stb_topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003 port map (
+              O_mac_counter_4_sva_en => cg_s_O_mac_counter_4_sva_en,
+              counter_nand_2_nl => counter_nand_2_nl,
+              counter_or_nl => counter_or_nl,
+              mux_14_cse => mux_14_cse,
+              for_1_for_1_for_1_for_1_nor_tmp => for_1_for_1_for_1_for_1_nor_tmp,
+              and_dcpl_3 => and_dcpl_3,
+              mux_tmp_25 => mux_tmp_25,
+              and_dcpl => and_dcpl_26,
+              for_1_unequal_3_tmp => for_1_unequal_3_tmp,
+              and_dcpl_64 => and_dcpl_64,
+              counter_nand_1_nl => counter_nand_1_nl,
+              counter_nand_3_nl => counter_nand_3_nl,
+              counter_nand_nl => counter_nand_nl,
+              W_mac_pntr_and_cse => W_mac_pntr_and_cse_1,
+              ppro_reset_CLOCKclk_out_port_14 => ppro_reset_CLOCKclk_out_port_15,
+              clk => clk_1,
+              rst => rst_1,
+              operator_5_false_1_acc_4_nl_5_5 => operator_5_false_1_acc_4_nl_5_5,
+              operator_5_false_1_acc_1_nl_5_5 => operator_5_false_1_acc_1_nl_5_5,
+              operator_5_false_1_acc_nl_5_5 => operator_5_false_1_acc_nl_5_5
+           );
   PROCESS (clk)
   BEGIN
     IF clk'EVENT AND ( clk = '1' ) THEN
@@ -8093,11 +8238,21 @@
         O_mac_pntr_sva <= STD_LOGIC_VECTOR'( "00000");
         W_irrel_cnt_sva_0 <= '0';
         I_irrel_cnt_sva_0 <= '0';
+       if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_4_sva <= STD_LOGIC_VECTOR'( "00000");
+       end if;
+       if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_3_sva <= STD_LOGIC_VECTOR'( "00000");
+       end if;
+       if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_2_sva <= STD_LOGIC_VECTOR'( "00000");
+       end if;
+       if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_1_sva <= STD_LOGIC_VECTOR'( "00000");
+       end if;
+       if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_0_sva <= STD_LOGIC_VECTOR'( "00000");
+       end if;
         W_mac_counter_4_sva <= STD_LOGIC_VECTOR'( "00000");
         W_mac_counter_3_sva <= STD_LOGIC_VECTOR'( "00000");
         W_mac_counter_2_sva <= STD_LOGIC_VECTOR'( "00000");
@@ -8114,16 +8269,26 @@
         O_mac_pntr_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_18;
         W_irrel_cnt_sva_0 <= (W_irrel_cnt_sva_dfm_mx0(0)) AND if_6_unequal_tmp_2;
         I_irrel_cnt_sva_0 <= (I_irrel_cnt_sva_dfm_mx0(0)) AND if_6_unequal_tmp_3;
+         if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_4_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_24(24
             DOWNTO 20);
+       end if;
+         if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_3_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_24(19
             DOWNTO 15);
+       end if;
+         if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_2_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_24(14
             DOWNTO 10);
+       end if;
+         if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_1_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_24(9
             DOWNTO 5);
+       end if;
+         if ( calypto_lib.powerpro_cg_package.is_enabled(cg_s_O_mac_counter_4_sva_en) ) then --PowerPro-CG
         O_mac_counter_0_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_24(4
             DOWNTO 0);
+       end if;
         W_mac_counter_4_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_29(24
             DOWNTO 20);
         W_mac_counter_3_sva <= libraries_O_addr_cnt_5_O_addr_type_L1_1_7448a59bcc55848f49259c21df88d8bc12ff1_29(19
@@ -9437,6 +9602,12 @@
   and_549_nl <= if_5_and_svs_1 AND (fsm_output(1));
   skid_buf_top_pop_1_mux_1_nl <= MUX_v_2_2_2(skid_buf_top_cnt_sva, skid_buf_top_cnt_sva_1_1,
       and_549_nl);
+  --PowerPro-CG
+  W_mac_pntr_and_cse_1 <= (W_mac_pntr_and_cse);
+  --PowerPro-CG
+  clk_1 <= (clk);
+  --PowerPro-CG
+  rst_1 <= (rst);
   z_out_3 <= STD_LOGIC_VECTOR(CONV_UNSIGNED(UNSIGNED(skid_buf_top_pop_1_mux_1_nl)
       + UNSIGNED'( "11"), 2));
 END v2;
@@ -10125,6 +10296,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
     IS
   PORT(
@@ -10153,15 +10326,19 @@
     I_instr_in_rsc_rdy : OUT STD_LOGIC;
     W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
     W_instr_in_rsc_vld : IN STD_LOGIC;
-    W_instr_in_rsc_rdy : OUT STD_LOGIC
-  );
+    W_instr_in_rsc_rdy : OUT STD_LOGIC;
+    --PowerPro-CG
+    ppro_reset_CLOCKclk_out_port_16 : in ieee.std_logic_1164.std_logic_vector(0 downto 0) );
+    
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
     IS
   -- Default Constants
 
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -10188,9 +10365,43 @@
       I_instr_in_rsc_rdy : OUT STD_LOGIC;
       W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
       W_instr_in_rsc_vld : IN STD_LOGIC;
-      W_instr_in_rsc_rdy : OUT STD_LOGIC
-    );
+      W_instr_in_rsc_rdy : OUT STD_LOGIC;
+    --PowerPro-CG
+    ppro_reset_CLOCKclk_out_port_15 : in ieee.std_logic_1164.std_logic_vector(0 downto 0) );
+      
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000003
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      O_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_wr_data_rsc_vld : IN STD_LOGIC;
+--      O_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      O_rd_data_rsc_dat : OUT STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_rd_data_rsc_vld : OUT STD_LOGIC;
+--      O_rd_data_rsc_rdy : IN STD_LOGIC;
+--      I_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      I_wr_data_rsc_vld : IN STD_LOGIC;
+--      I_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      W_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_wr_data_rsc_vld : IN STD_LOGIC;
+--      W_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      wr_data_zero_guard_rsc_dat : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_vld : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_rdy : OUT STD_LOGIC;
+--      O_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      O_instr_in_rsc_vld : IN STD_LOGIC;
+--      O_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      I_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      I_instr_in_rsc_vld : IN STD_LOGIC;
+--      I_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      W_instr_in_rsc_vld : IN STD_LOGIC;
+--      W_instr_in_rsc_rdy : OUT STD_LOGIC
+--    );
+--  END COMPONENT;
+                
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_inst_O_wr_data_rsc_dat
       : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_inst_O_rd_data_rsc_dat
@@ -10235,8 +10446,10 @@
       I_instr_in_rsc_rdy => I_instr_in_rsc_rdy,
       W_instr_in_rsc_dat => rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_inst_W_instr_in_rsc_dat,
       W_instr_in_rsc_vld => W_instr_in_rsc_vld,
-      W_instr_in_rsc_rdy => W_instr_in_rsc_rdy
-    );
+      W_instr_in_rsc_rdy =>                             W_instr_in_rsc_rdy,
+      --PowerPro-CG
+      ppro_reset_CLOCKclk_out_port_15 => ppro_reset_CLOCKclk_out_port_16 );
+      
   rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_inst_O_wr_data_rsc_dat
       <= O_wr_data_rsc_dat;
   O_rd_data_rsc_dat <= rf_5_32_32_32_1_1_1_1_1_1_O_partial_type_O_addr_type_L1_I_type_I_addr_type_L1_W_type_W_addr_type_L1_run_inst_O_rd_data_rsc_dat;
@@ -10265,6 +10478,8 @@
 USE mgc_hls.topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000151.ALL;
 
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ENTITY topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
     IS
   PORT(
@@ -10304,10 +10519,14 @@
     layer_instruction_in_rsc_dat_relevancy_8 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
     layer_instruction_in_rsc_dat_bound_8 : IN STD_LOGIC_VECTOR (69 DOWNTO 0);
     layer_instruction_in_rsc_vld : IN STD_LOGIC;
-    layer_instruction_in_rsc_rdy : OUT STD_LOGIC
-  );
+    layer_instruction_in_rsc_rdy : OUT STD_LOGIC;
+    --PowerPro-CG
+    ppro_reset_CLOCKclk_out_port_17 : in ieee.std_logic_1164.std_logic_vector(0 downto 0) );
+    
 END topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
     IS
   -- Default Constants
@@ -10390,7 +10609,7 @@
   SIGNAL CCInst_W_instr_L2_out_rsc_dat : STD_LOGIC_VECTOR (109 DOWNTO 0);
   SIGNAL CCInst_W_instr_L3_out_rsc_dat : STD_LOGIC_VECTOR (159 DOWNTO 0);
 
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -10417,9 +10636,43 @@
       I_instr_in_rsc_rdy : OUT STD_LOGIC;
       W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
       W_instr_in_rsc_vld : IN STD_LOGIC;
-      W_instr_in_rsc_rdy : OUT STD_LOGIC
-    );
+      W_instr_in_rsc_rdy : OUT STD_LOGIC;
+    --PowerPro-CG
+    ppro_reset_CLOCKclk_out_port_16 : in ieee.std_logic_1164.std_logic_vector(0 downto 0) );
+      
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000002
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      O_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_wr_data_rsc_vld : IN STD_LOGIC;
+--      O_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      O_rd_data_rsc_dat : OUT STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_rd_data_rsc_vld : OUT STD_LOGIC;
+--      O_rd_data_rsc_rdy : IN STD_LOGIC;
+--      I_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      I_wr_data_rsc_vld : IN STD_LOGIC;
+--      I_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      W_wr_data_rsc_dat : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_wr_data_rsc_vld : IN STD_LOGIC;
+--      W_wr_data_rsc_rdy : OUT STD_LOGIC;
+--      wr_data_zero_guard_rsc_dat : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_vld : IN STD_LOGIC;
+--      wr_data_zero_guard_rsc_rdy : OUT STD_LOGIC;
+--      O_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      O_instr_in_rsc_vld : IN STD_LOGIC;
+--      O_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      I_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      I_instr_in_rsc_vld : IN STD_LOGIC;
+--      I_instr_in_rsc_rdy : OUT STD_LOGIC;
+--      W_instr_in_rsc_dat : IN STD_LOGIC_VECTOR (49 DOWNTO 0);
+--      W_instr_in_rsc_vld : IN STD_LOGIC;
+--      W_instr_in_rsc_rdy : OUT STD_LOGIC
+--    );
+--  END COMPONENT;
+                
   SIGNAL PE_O_wr_data_rsc_dat : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL PE_O_rd_data_rsc_dat : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL PE_I_wr_data_rsc_dat : STD_LOGIC_VECTOR (15 DOWNTO 0);
@@ -10511,8 +10764,10 @@
       I_instr_in_rsc_rdy => I_instr_in_rsc_rdy_nPE_bud,
       W_instr_in_rsc_dat => PE_W_instr_in_rsc_dat,
       W_instr_in_rsc_vld => W_instr_L1_out_rsc_vld_nCCInst_bud,
-      W_instr_in_rsc_rdy => W_instr_in_rsc_rdy_nPE_bud
-    );
+      W_instr_in_rsc_rdy =>                             W_instr_in_rsc_rdy_nPE_bud,
+      --PowerPro-CG
+      ppro_reset_CLOCKclk_out_port_16 => ppro_reset_CLOCKclk_out_port_17 );
+      
   PE_O_wr_data_rsc_dat <= O_data_in_rsc_dat_data;
   O_rd_data_rsc_dat_nPE <= PE_O_rd_data_rsc_dat;
   PE_I_wr_data_rsc_dat <= I_data_in_rsc_dat_data;
@@ -10568,6 +10823,8 @@
   );
 END top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000;
 
+library ieee, calypto_lib;
+use calypto_lib.top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_calypto_package.all;
 ARCHITECTURE v2 OF top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000
     IS
   -- Default Constants
@@ -10576,7 +10833,7 @@
   -- Interconnect Declarations
   SIGNAL O_data_out_rsc_dat_data : STD_LOGIC_VECTOR (15 DOWNTO 0);
 
-  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
+  component topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
     PORT(
       clk : IN STD_LOGIC;
       rst : IN STD_LOGIC;
@@ -10614,9 +10871,54 @@
       layer_instruction_in_rsc_dat_relevancy_8 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
       layer_instruction_in_rsc_dat_bound_8 : IN STD_LOGIC_VECTOR (69 DOWNTO 0);
       layer_instruction_in_rsc_vld : IN STD_LOGIC;
-      layer_instruction_in_rsc_rdy : OUT STD_LOGIC
-    );
+      layer_instruction_in_rsc_rdy : OUT STD_LOGIC;
+    --PowerPro-CG
+    ppro_reset_CLOCKclk_out_port_17 : in ieee.std_logic_1164.std_logic_vector(0 downto 0) );
+      
   END COMPONENT;
+                
+--  COMPONENT topless_5comma_32comma_1comma_1comma_O_addr_type_L1comma_32comma_1comma_1comma_I_addr_type_L1comma_32comma_1comma_1comma_W_addr_type_L1comma_5comma_288comma_8comma_8comma_O_addr_type_L2comma_312comma_8comma_8comma_I_addr_type_L2c000000
+--    PORT(
+--      clk : IN STD_LOGIC;
+--      rst : IN STD_LOGIC;
+--      O_data_in_rsc_dat_data : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_data_in_rsc_vld : IN STD_LOGIC;
+--      O_data_in_rsc_rdy : OUT STD_LOGIC;
+--      O_data_out_rsc_dat_data : OUT STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      O_data_out_rsc_vld : OUT STD_LOGIC;
+--      O_data_out_rsc_rdy : IN STD_LOGIC;
+--      I_data_in_rsc_dat_data : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      I_data_in_rsc_vld : IN STD_LOGIC;
+--      I_data_in_rsc_rdy : OUT STD_LOGIC;
+--      W_data_in_rsc_dat_data : IN STD_LOGIC_VECTOR (15 DOWNTO 0);
+--      W_data_in_rsc_vld : IN STD_LOGIC;
+--      W_data_in_rsc_rdy : OUT STD_LOGIC;
+--      zero_guard_in_rsc_dat : IN STD_LOGIC;
+--      zero_guard_in_rsc_vld : IN STD_LOGIC;
+--      zero_guard_in_rsc_rdy : OUT STD_LOGIC;
+--      layer_instruction_in_rsc_dat_relevancy : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound : IN STD_LOGIC_VECTOR (24 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_1 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_1 : IN STD_LOGIC_VECTOR (54 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_2 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_2 : IN STD_LOGIC_VECTOR (79 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_3 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_3 : IN STD_LOGIC_VECTOR (24 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_4 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_4 : IN STD_LOGIC_VECTOR (44 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_5 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_5 : IN STD_LOGIC_VECTOR (69 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_6 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_6 : IN STD_LOGIC_VECTOR (24 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_7 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_7 : IN STD_LOGIC_VECTOR (44 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_relevancy_8 : IN STD_LOGIC_VECTOR (4 DOWNTO 0);
+--      layer_instruction_in_rsc_dat_bound_8 : IN STD_LOGIC_VECTOR (69 DOWNTO 0);
+--      layer_instruction_in_rsc_vld : IN STD_LOGIC;
+--      layer_instruction_in_rsc_rdy : OUT STD_LOGIC
+--    );
+--  END COMPONENT;
+                
   SIGNAL top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000002
       : STD_LOGIC_VECTOR (15 DOWNTO 0);
   SIGNAL top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000003
@@ -10661,6 +10963,10 @@
       : STD_LOGIC_VECTOR (4 DOWNTO 0);
   SIGNAL top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000023
       : STD_LOGIC_VECTOR (69 DOWNTO 0);
+  --PowerPro-CG
+  signal ppro_reset_CLOCKclk_out_port_14 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
+  --PowerPro-CG
+  signal rst_1 : ieee.std_logic_1164.std_logic_vector(0 downto 0);
 
 BEGIN
   top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000001
@@ -10702,8 +11008,10 @@
       layer_instruction_in_rsc_dat_relevancy_8 => top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000022,
       layer_instruction_in_rsc_dat_bound_8 => top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000023,
       layer_instruction_in_rsc_vld => layer_instruction_in_rsc_vld,
-      layer_instruction_in_rsc_rdy => layer_instruction_in_rsc_rdy
-    );
+      layer_instruction_in_rsc_rdy =>                                       layer_instruction_in_rsc_rdy,
+      --PowerPro-CG
+      ppro_reset_CLOCKclk_out_port_17 => ppro_reset_CLOCKclk_out_port_14 );
+      
   top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000002
       <= O_data_in_rsc_dat;
   O_data_out_rsc_dat_data <= top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000003;
@@ -10748,6 +11056,13 @@
   top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000023
       <= layer_instruction_in_rsc_dat(69 DOWNTO 0);
 
+  --PowerPro-CG
+  rst_1 <= (rst);
+  --PowerPro-CG
+  top_5_32_1_1_O_addr_type_L1_32_1_1_I_add : top_5_32_1_1_O_addr_type_L1_32_1_1_I_addr_type_L1_32_1_1_W_addr_type_L1_5_288_8_8_O_addr_type_L2_312_8_8_I_addr_type_L2_1728_8_8_W_addr_type_L2_5_8640_8_8_O_addr_type_L3_8640_8_8_I_addr_type_L3_55296_8_8_W_addr_type_L3_O_partial_000000_powerpro_reset_mod port map (
+              rst => rst_1,
+              ppro_reset_CLOCKclk_out_port => ppro_reset_CLOCKclk_out_port_14
+           );
   O_data_out_rsc_dat <= O_data_out_rsc_dat_data;
 END v2;
 
